matplotlib fails to detect Tk if path is different from Tcl on Cray

Problem is that setupext.py from matplotlib is guessing Tk path based on Tcl location:
```
# Guess Tk location based on Tcl location
(head, tail) = os.path.split(tcl_lib_dir)
tail = tail.replace('Tcl', 'Tk').replace('tcl', 'tk')
````
If, for example, the Tk easyconfig file adds versionsuffix = '-no-X11', the replace is not working and matplotlib fails to detect Tk.

I removed no-X11 versionsuffix as workaround, but it would be nice to understand if Tk is working correctly and how the detection is working in other builds.

More info here https://github.com/hpcugent/easybuild-easyconfigs/pull/2608

Same problem here, foss-2016a on SLES11 SP3. Trying to compile matplotlib 1.4.3 and 1.5.1 results in the same error:

/apps/easybuild/software/shadowfax-westmere/binutils/2.25-GCCcore-4.9.3/bin/ld.gold: error: cannot find -ltcl
/apps/easybuild/software/shadowfax-westmere/binutils/2.25-GCCcore-4.9.3/bin/ld.gold: error: cannot find -ltk

After compiling a new Tk without the version suffix (Tk/8.6.4-foss-2016a), matplotlib was able to compile.

--- a/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2015b-Python-2.7.10.eb
+++ b/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2015b-Python-2.7.10.eb
@@ -28,6 +28,7 @@ exts_list = [
     }),
     (name, version, {
         'source_urls': ['https://pypi.python.org/packages/source/m/matplotlib'],
+        'patches': ['matplotlib-1.5.1_fix-Tcl-Tk-libdir.patch'],
     }),
 ]
 

--- a/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2016a-Python-2.7.11.eb
+++ b/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-foss-2016a-Python-2.7.11.eb
@@ -28,6 +28,7 @@ exts_list = [
     }),
     (name, version, {
         'source_urls': ['https://pypi.python.org/packages/source/m/matplotlib'],
+        'patches': ['matplotlib-1.5.1_fix-Tcl-Tk-libdir.patch'],
     }),
 ]
 

--- a/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-intel-2015b-Python-2.7.11.eb
+++ b/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-intel-2015b-Python-2.7.11.eb
@@ -28,6 +28,7 @@ exts_list = [
     }),
     (name, version, {
         'source_urls': ['https://pypi.python.org/packages/source/m/matplotlib'],
+        'patches': ['matplotlib-1.5.1_fix-Tcl-Tk-libdir.patch'],
     }),
 ]
 

--- a/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-intel-2016a-Python-2.7.11.eb
+++ b/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-intel-2016a-Python-2.7.11.eb
@@ -28,6 +28,7 @@ exts_list = [
     }),
     (name, version, {
         'source_urls': ['https://pypi.python.org/packages/source/m/matplotlib'],
+        'patches': ['matplotlib-1.5.1_fix-Tcl-Tk-libdir.patch'],
     }),
 ]
 

--- a/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-intel-2016a-Python-3.5.1.eb
+++ b/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1-intel-2016a-Python-3.5.1.eb
@@ -29,6 +29,7 @@ exts_list = [
     }),
     (name, version, {
         'source_urls': ['https://pypi.python.org/packages/source/m/matplotlib'],
+        'patches': ['matplotlib-1.5.1_fix-Tcl-Tk-libdir.patch'],
     }),
 ]
 

--- a/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1_fix-Tcl-Tk-libdir.patch
+++ b/easybuild/easyconfigs/m/matplotlib/matplotlib-1.5.1_fix-Tcl-Tk-libdir.patch
@@ -1,0 +1,26 @@
+ensure that Tcl/Tk library paths are correct, to fix segmentation fault that occurs when plotting because system
+libraries are picked up;
+the problem is that the query_tcktl method guesses the Tk library path based on the Tcl library path by simply
+replacing 'Tcl'/'tcl' with 'Tk'/'tk', which doesn't work if the Tk install prefix is different (e.g. due to a
+versionsuffix)
+author: Kenneth Hoste (HPC-UGent)
+-- matplotlib-1.5.1.orig/setupext.py	2016-01-11 04:06:08.000000000 +0100
++++ matplotlib-1.5.1/setupext.py	2016-04-29 15:43:21.280170000 +0200
+@@ -1560,7 +1560,16 @@
+ 
+             # Query Tcl/Tk system for library paths and version string
+             try:
+-                tcl_lib_dir, tk_lib_dir, tk_ver = self.query_tcltk()
++                tcl_lib_dir, tk_lib_dir, tk_ver = None, None, None
++                if 'EBROOTTCL' in os.environ and 'EBVERSIONTCL' in os.environ:
++                    tcl_ver = '.'.join(os.environ['EBVERSIONTCL'].split('.')[:2])
++                    tcl_lib_dir = os.path.join(os.environ['EBROOTTCL'], 'lib', 'tcl%s' % tcl_ver)
++                if 'EBROOTTK' in os.environ and 'EBVERSIONTK' in os.environ:
++                    tk_ver = '.'.join(os.environ['EBVERSIONTK'].split('.')[:2])
++                    tk_lib_dir = os.path.join(os.environ['EBROOTTK'], 'lib', 'tk%s' % tk_ver)
++
++                if tcl_lib_dir is None or tk_lib_dir is None or tk_ver is None:
++                        tcl_lib_dir, tk_lib_dir, tk_ver = self.query_tcltk()
+             except:
+                 tk_ver = ''
+                 result = self.hardcoded_tcl_config()

